Jint/Runtime/TypeConverter.cs

```diff
@@ -246,7 +250,40 @@ public static uint ToUint32(JsValue o)
         /// <returns></returns>
         public static ushort ToUint16(JsValue o)
         {
-            return (ushort)(uint)ToNumber(o);
+            return (ushort) (uint) ToNumber(o);
+        }
+
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string ToString(long i)
+        {
+            return i >= 0 && i < intToString.Length
+                ? (intToString[i] = intToString[i] ?? i.ToString())
+                : i.ToString();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string ToString(int i)
+        {
+            return i >= 0 && i < intToString.Length
+                ? (intToString[i] = intToString[i] ?? i.ToString())
+                : i.ToString();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string ToString(uint i)
+        {
+            return i >= 0 && i < intToString.Length
+                ? (intToString[i] = intToString[i] ?? i.ToString())
+                : i.ToString();
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static string ToString(char c)
+        {
+            return c >= 0 && c < charToString.Length
+                ? (charToString[c] = charToString[c] ?? c.ToString())
```

Comment:
I believe this pattern is not thread safe as the array is shared, and any thread could write to it if it's the first to encounter the number. Unless you would initialize it completely in the static constructor and just read it here.

10 thousand integers of 5 digits max to string would take around 100KB of memory I assume.